cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -306,7 +307,43 @@ func execute_code{
         ),
     );
 
-    // TODO: Handle precompiled contracts
+    // code_address might be optional in create scenarios at this point.
+    if (cast(evm.value.message.value.code_address.value, felt) != 0) {
+        let (precompile_address, precompile_fn) = precompile_table_lookup(
+            [evm.value.message.value.code_address.value]
+        );
+        // Addresses that are not precompiles return 0.
+        if (precompile_address != 0) {
+            // Prepare arguments
+            [ap] = range_check_ptr, ap++;
+            [ap] = range_check_ptr, ap++;
+            [ap] = bitwise_ptr, ap++;
+            [ap] = keccak_ptr, ap++;
+            [ap] = evm.value, ap++;
+
+            call abs precompile_fn;
+
+            let range_check_ptr = [ap - 5];
+            let bitwise_ptr = cast([ap - 4], BitwiseBuiltin*);
+            let keccak_ptr = cast([ap - 3], KeccakBuiltin*);
+            let evm_ = cast([ap - 2], EvmStruct*);
+            let err = cast([ap - 1], EthereumException*);
+            tempvar evm = Evm(evm_);
+
+            if (cast(err, felt) != 0) {
+                EvmImpl.set_gas_left{evm=evm}(Uint(0));
```

Comment:
```suggestion
                EvmImpl.set_gas_left(Uint(0));
```
Is this necessary?

Reply:
yes because evm is not an implicit arg of the function

Reply:
oh right